src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -147,8 +159,32 @@ def wrapper(*args, **kwargs):
                 "Project definition could not be found. The nativeapp_definition_v2_to_v1 command decorator assumes with_project_definition() was called before it."
             )
         if original_pdf.definition_version == "2":
-            pdfv1 = _pdf_v2_to_v1(original_pdf)
+            package_entity_id = kwargs.get("package_entity_id", "")
+            app_entity_id = kwargs.get("app_entity_id", "")
+            pdfv1 = _pdf_v2_to_v1(original_pdf, package_entity_id, app_entity_id)
             get_cli_context_manager().set_project_definition(pdfv1)
         return func(*args, **kwargs)
 
-    return wrapper
+    return _options_decorator_factory(
```

Comment:
This is great -- I like how it allows us to keep all the v2 -> v1 stuff in one place.

However, I can see myself getting tripped up by needing to provide both. Can we infer the pkg from the app when only the app is provided (i.e. by introspecting the `from` clause)?

Reply:
yeah that's a good idea

Reply:
done